src/browser_action/on-each-interaction.js

```diff
@@ -0,0 +1,58 @@
+/*
+ Copyright 2023 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+/**
+ * @param {Function} callback
+ */
+export function onEachInteraction(callback) {
+  const valueToRating = (score) => score <= 200 ? 'good' : score <= 500 ? 'needs-improvement' : 'poor';
+
+  const observer = new PerformanceObserver((list) => {
+    const interactions = {};
+
+    for (const entry of list.getEntries().filter((entry) => entry.interactionId)) {
+      interactions[entry.interactionId] = interactions[entry.interactionId] || [];
+      interactions[entry.interactionId].push(entry);
+    }
+
+    // Will report as a single interaction even if parts are in separate frames.
+    // Consider splitting by animation frame.
+    for (const interaction of Object.values(interactions)) {
+      const entry = interaction.reduce((prev, curr) => prev.duration >= curr.duration ? prev : curr);
+      const value = entry.duration;
+
+      callback({
+        attribution: {
+          eventEntry: entry,
+          eventTarget: '',
+          eventTime: entry.startTime,
+          eventType: entry.name,
+          loadState: 'unknown',
+        },
+        delta: 0,
+        entries: interaction,
+        id: 'none',
+        name: 'Interaction',
+        navigationType: 'unknown',
+        rating: valueToRating(value),
+        value,
```

Comment:
Let's some of these that won't be set. I don't think it needs to be exactly the same as INP entry and worry about people expecting these to be set.

```suggestion
      callback({
        attribution: {
          eventEntry: entry,
          eventTime: entry.startTime,
          eventType: entry.name,
        },
        entries: interaction,
        name: 'Interaction',
        rating: valueToRating(value),
        value,
```